packages/realm-react/src/UserProvider.tsx

```diff
@@ -35,30 +35,52 @@ type UserProviderProps = {
   children: React.ReactNode;
 };
 
+function userWasUpdated(userA: Realm.User | null, userB: Realm.User | null) {
+  if (!userA && !userB) {
+    return false;
+  } else if (userA && userB) {
+    return (
+      userA.id !== userB.id ||
+      userA.state !== userB.state ||
+      userA.accessToken !== userB.accessToken ||
+      userA.refreshToken !== userB.refreshToken
+    );
+  } else {
+    return true;
+  }
+}
+
 /**
  * React component providing a Realm user on the context for the sync hooks
  * to use. A `UserProvider` is required for an app to use the hooks.
  */
 export const UserProvider: React.FC<UserProviderProps> = ({ fallback: Fallback, children }) => {
   const app = useApp();
   const [user, setUser] = useState<Realm.User | null>(() => app.currentUser);
+  const [, forceUpdate] = useReducer((x) => x + 1, 0);
 
   // Support for a possible change in configuration
-  if (app.currentUser?.id != user?.id) {
-    setUser(app.currentUser);
+  const currentUser = app.currentUser;
```

Comment:
Perhaps better to wrap this in a useEffect like:
```
useEffect(() => {
  const currentUser = app.currentUser;
  if (userWasUpdated(user, currentUser)) {
    setUser(currentUser);
  }
}, [app.currentUser, user]);
```

Reply:
Thanks for you input 🙂 In this case, we intentionally want to avoid the `useEffect()` so as to not render stale state. Checking if the user was updated outside of the `useEffect()` and calling `setUser()` there will allow the rerender to restart without also having to rerender the children using the stale state. (You can read more about that [here](https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes).)